src/hydra-queue-runner/hydra-queue-runner.cc

```diff
@@ -449,74 +463,131 @@ bool State::checkCachedFailure(Step::ptr step, Connection & conn)
     return false;
 }
 
-
 void State::notificationSender()
 {
-    while (true) {
-        try {
+  using namespace std::string_literals;
+  std::optional<hydra_notify_state> hydra_notify = std::nullopt;
+
+  while (true) {
```

Comment:
Is there any maximum number of times this respawns? I would include an increasing sleep between respawns to keep from using 100% CPU respawning processes that immediately fail in case of situations where there is some reason it will never succeed, unless I'm misreading the code.